TeachingRecordSystem/src/TeachingRecordSystem.Api/V3/Implementation/Operations/SetProfessionalStatus.cs

```diff
@@ -0,0 +1,610 @@
+            if (!IsSuccess)
+            {
+                return ApiError.InvalidTrainingCountryReference(command.TrainingCountryReference);
+            }
+
+            countryId = Result!.Id;
+        }
+
+        Guid? providerId = null;
+        if (command.TrainingProviderUkprn is not null)
+        {
+            var (IsSuccess, Result) = await command.TrainingProviderUkprn.TryConvertFromUkPrnAsync(referenceDataCache);
+            if (!IsSuccess)
+            {
+                return ApiError.InvalidTrainingProviderUkprn(command.TrainingProviderUkprn);
+            }
+
+            providerId = Result!.Id;
+        }
+
+        if (command.RouteTypeId.IsOverseas())
+        {
+            providerId = await DeriveIttProviderForOverseasQualifiedTeacherAsync(command.RouteTypeId);
+        }
+
+        Guid? ittQualificationId = null;
+        if (command.DegreeTypeId is not null)
+        {
+            var (IsSuccess, Result) = await command.DegreeTypeId.Value.TryConvertFromTrsDegreeTypeIdAsync(referenceDataCache);
+            if (!IsSuccess)
+            {
+                return ApiError.InvalidDegreeType(command.DegreeTypeId.Value);
+            }
+
+            ittQualificationId = Result!.Id;
+        }
+
+        var itt = lookupData.Itt;
+        var isNewItt = itt is null;
+        var cohortYear = command.TrainingEndDate?.Year.ToString();
+        var existingProgrammeType = itt?.dfeta_ProgrammeType;
+        if (isNewItt)
+        {
+            itt = new dfeta_initialteachertraining
+            {
+                Id = Guid.NewGuid(),
+                dfeta_PersonId = dqtContact.Id.ToEntityReference(Contact.EntityLogicalName),
+                dfeta_SlugId = command.SourceApplicationReference,
+                dfeta_TraineeID = dqtContact.dfeta_HUSID
+            };
+        }
+        else
+        {
+            // No current provision for changing overseas
+            if (command.RouteTypeId.IsOverseas())
+            {
+                return ApiError.UpdatesNotAllowedForRouteType(command.RouteTypeId);
+            }
+
+            // Can't change between Early Years and non-Early Years
+            if (isEarlyYears != existingProgrammeType?.IsEarlyYears())
+            {
+                return ApiError.UnableToChangeRouteType();
+            }
+
+            switch (itt!.dfeta_Result)
+            {
+                // If the route has already been Awarded then this can't be changed via the API - needs to be altered via TRS console
+                case dfeta_ITTResult.Pass:
+                    return ApiError.RouteToProfessionalStatusAlreadyAwarded();
+                case dfeta_ITTResult.Fail:
+                    switch (command.Status)
+                    {
+                        case ProfessionalStatusStatus.Failed:
+                            return new SetProfessionalStatusResult(); //No-op
```

Comment:
What about if other attributes have been changed?

Reply:
this is just replicating what the v2 API does

https://github.com/DFE-Digital/teaching-record-system/blob/main/TeachingRecordSystem/src/TeachingRecordSystem.Core/Dqt/DataverseAdapter.UpdateTeacher.cs#L58